.gitmodules

```diff
@@ -62,3 +62,6 @@
 [submodule "metagraph/external-libraries/Simple-Web-Server"]
 	path = metagraph/external-libraries/Simple-Web-Server
 	url = https://gitlab.com/eidheim/Simple-Web-Server.git
+[submodule "metagraph/external-libraries/folly"]
+	path = metagraph/external-libraries/folly
+	url = https://github.com/facebook/folly
```

Comment:
Since it's optional, maybe we should download it in cmake, like googletests?

Reply:
googletests is a submodule, so it's downloaded every time. also, Folly only takes up 19MB, so it's not that big of a download.

I think making this change will add unnecessary complexity.